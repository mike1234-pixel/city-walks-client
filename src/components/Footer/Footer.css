.footer {
  background-color: #212121;
}

/* removes bootstrap margin on footer */
.mt-4 {
  margin-top: 0 !important;
}

.footer-ul {
  list-style: none;
  padding-left: 0;
}
