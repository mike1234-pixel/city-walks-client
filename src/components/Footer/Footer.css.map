{"version":3,"sourceRoot":"","sources":["Footer.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;;;AAGF;AACA;EACE;;;AAGF;EACE;EACA;;;AAGF;AAAA;AAAA;EAGE;;;AAGF;EACE;AAAA;IAEE;IACA;;;AAIJ;EACE;;AACA;EACE","file":"Footer.css"}