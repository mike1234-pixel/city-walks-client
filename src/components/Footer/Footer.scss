/** @format */

.footer {
  background-color: #212121;
}

/* removes bootstrap margin on footer */
.mt-4 {
  margin-top: 0 !important;
}

.footer-ul {
  list-style: none;
  padding-left: 0;
}

.footer-ul > li > a:hover,
.footer-ul > li > a:focus,
.footer-ul > li > a:active {
  color: var(--blue);
}

@media (max-width: 777px) {
  .footer-site-links,
  .footer-extra-links {
    visibility: hidden;
    display: none;
  }
}
