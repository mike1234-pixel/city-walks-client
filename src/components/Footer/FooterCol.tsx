import React, { ReactNode } from "react";

interface Props {
  children: ReactNode;
  title: string;
}

const FooterCol: React.FC<Props> = (props: Props) => {
  const { title } = props;

  return (
    <>
      <h5 className="title">{title}</h5>
      <ul className="footer-ul">{props.children}</ul>
    </>
  );
};

export default FooterCol;
