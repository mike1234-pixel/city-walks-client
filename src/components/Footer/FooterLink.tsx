import React from "react";
import { Link } from "react-router-dom";

interface Props {
  path: string;
  title: string;
}

const FooterLink: React.FC<Props> = (props: Props) => {
  const { path, title } = props;

  return (
    <li className="list-unstyled">
      <Link to={path}>{title}</Link>
    </li>
  );
};

export default FooterLink;
