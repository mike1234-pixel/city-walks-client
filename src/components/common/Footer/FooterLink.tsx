import { Link } from "react-router-dom";

interface FooterLinkProps {
  path: string;
  title: string;
}

const FooterLink = (props: FooterLinkProps) => {
  const { path, title } = props;

  return (
    <li className="list-unstyled">
      <Link to={path}>{title}</Link>
    </li>
  );
};

export default FooterLink;
