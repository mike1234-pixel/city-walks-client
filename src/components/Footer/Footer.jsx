import React, { useContext } from 'react'
import { Link } from "react-router-dom"
import { MDBCol, MDBContainer, MDBRow, MDBFooter } from 'mdbreact'
import { LoginContext } from "../../context/LoginContext"
import axios from "axios"
import './Footer.css'

const Footer = () => {

  const { loggedIn, userId, logOut } = useContext(LoginContext)

  const deleteAccount = () => {

    const payload = {
      userId: userId,
    }

    axios.delete("http://localhost:5000/delete-account", {data: payload})
        .then((err) => {
            if (err) {
              console.log(err);
            }
          });
  
          alert("Account Deleted. You can sign up again at any time.")
          logOut()
          window.scrollTo(0, 0);
  }

  return (
    <MDBFooter className="font-small pt-4 mt-4 footer">
      <MDBContainer fluid className="text-center text-md-left">
        <MDBRow>
          <MDBCol md="6">
            <h5 className="title">City Walks is...</h5>
            <p>
              (City Walks description)
            </p>
          </MDBCol>
          <MDBCol md="3">
            <h5 className="title">Site Links</h5>
            <ul className="footer-ul">
            <li className="list-unstyled">
                <Link to="/">Home</Link>
              </li>
              <li className="list-unstyled">
                <Link to="/cities">Cities</Link>
              </li>
              <li className="list-unstyled">
                <Link to="/walks">Walks</Link>
              </li>
              <li className="list-unstyled">
                <Link to="/about">About</Link>
              </li>
              <li className="list-unstyled">
                <Link to="/blog">Blog</Link>
              </li>
              <li className="list-unstyled">
                <Link to="/contact">Contact</Link>
              </li>
              <li className="list-unstyled">
                <Link to="/boards">User Login</Link>
              </li>
            </ul>
          </MDBCol>
          <MDBCol md="3">
          <h5 className="title">Extra Links</h5>
          <ul className="footer-ul">
            <li className="list-unstyled">
                <Link to="/admin">Admin portal</Link>
            </li>
            <li className="list-unstyled">
                <Link to="/privacy">Privacy Policy</Link>
            </li>
            {loggedIn && 
            <li className="list-unstyled">
                <a onClick={deleteAccount}>Delete Account</a>
            </li>}
            </ul>
          </MDBCol>
        </MDBRow>
      </MDBContainer>
      <div className="footer-copyright text-center py-3">
        <MDBContainer fluid>
          &copy; {new Date().getFullYear()} Copyright
        </MDBContainer>
      </div>
    </MDBFooter>
  );
}

export default Footer;