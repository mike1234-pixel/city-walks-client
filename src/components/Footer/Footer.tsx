import React from "react"
import { useHistory } from "react-router-dom"
import { MDBCol, MDBContainer, MDBRow, MDBFooter } from "mdbreact"
import { GiWalkingBoot } from "react-icons/gi"
import axios, { AxiosError, AxiosResponse } from "axios"
import { connect } from "react-redux"
import { Dispatch, bindActionCreators } from "redux"
import RootState from "../../types/State/Root/State"
import Account from "../../types/PostRequests/Account"
import {
  setLoggedIn,
  setUserFirstName,
  setUserId,
  setUserLastName,
} from "../../actions/actions"
import "./Footer.css"
import FooterLink from "./FooterLink"
import FooterCol from "./FooterCol"
import * as Actions from "../../types/Actions"

interface Props {
  loggedIn: boolean;
  userId: string;
  setLoggedIn: (loggedIn: boolean) => Actions.SetLoggedIn;
  setUserId: (userId: string) => Actions.SetUserId;
  setUserFirstName: (userFirstName: string) => Actions.SetUserFirstName;
  setUserLastName: (userLastName: string) => Actions.SetUserLastName;
}

const Footer: React.FC<Props> = (props: Props) => {
  const {
    loggedIn,
    userId,
    setLoggedIn,
    setUserId,
    setUserFirstName,
    setUserLastName,
  } = props

  const pushSlug: Function = useHistory().push

  const logOut = () => {
    localStorage.clear()
    localStorage.setItem("popupVisible", "false")
    setLoggedIn(false)
    setUserId("")
    setUserFirstName("")
    setUserLastName("")
    alert("Logged out successfully.")
    pushSlug("/forum")
  }

  const deleteAccount = () => {
    const payload: Account = {
      userId: userId,
    }

    axios
      .delete("https://city-walks-production.up.railway.app/delete-account", {
        data: payload,
      })
      .then((res: AxiosResponse) => {
        console.log("account deleted")
      })
      .catch((err: AxiosError) => {
        console.log(err, "account not deleted, try again.")
      })

    alert("Account Deleted. You can sign up again at any time.")
    logOut()
    window.scrollTo(0, 0)
  }

  return (
    <MDBFooter className='font-small pt-4 mt-4 footer'>
      <MDBContainer className='text-left'>
        <MDBRow>
          <MDBCol md='3'>
            <h5 className='title'>
              City Walks... <GiWalkingBoot />
            </h5>
            <p>
              City Walks is a website that publishes illustrated guided walks,
              with routes, maps and local attractions to explore.
            </p>
          </MDBCol>
          <MDBCol md='3' className='offset-md-3 footer-site-links'>
            <FooterCol title='Site Links'>
              <FooterLink path='/' title='Home' />
              <FooterLink path='/cities' title='Cities' />
              <FooterLink path='/walks' title='Walks' />
              <FooterLink path='/about' title='About' />
              <FooterLink path='/sights' title='Sights' />
              <FooterLink path='/contact' title='Contact' />
              <FooterLink path='/forum/login' title='User Login' />
            </FooterCol>
          </MDBCol>
          <MDBCol md='3' className='footer-extra-links'>
            <FooterCol title='Extra Links'>
              <FooterLink path='/admin' title='Admin Portal' />
              <FooterLink path='/privacy' title='Privacy Policy' />
              {loggedIn && (
                <li className='list-unstyled'>
                  <a onClick={deleteAccount}>Delete Account</a>
                </li>
              )}
            </FooterCol>
          </MDBCol>
        </MDBRow>
      </MDBContainer>
      <div className='footer-copyright text-center py-3'>
        <MDBContainer fluid>
          &copy; {new Date().getFullYear()} Copyright{" "}
          <a
            href='https://github.com/mike1234-pixel'
            target='_blank'
            className='footer-github-link'
          >
            Michael Tandy
          </a>
        </MDBContainer>
      </div>
    </MDBFooter>
  )
}

const mapStateToProps: (state: RootState) => void = (state) => ({
  loggedIn: state.loginState.loggedIn,
  userId: state.loginState.userId,
})

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    dispatch,
    ...bindActionCreators(
      { setLoggedIn, setUserId, setUserFirstName, setUserLastName },
      dispatch
    ),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Footer)
