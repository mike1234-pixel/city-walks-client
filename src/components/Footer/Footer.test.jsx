import renderer from 'react-test-renderer'
import { BrowserRouter } from "react-router-dom"
import { LoginContextProvider } from "../../context/LoginContext"
import Footer from './Footer'


// snapshot test
it('Footer renders correctly', () => {  

  const tree = renderer
    .create(
    <LoginContextProvider>
      <BrowserRouter>
        <Footer/>
      </BrowserRouter>
    </LoginContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});