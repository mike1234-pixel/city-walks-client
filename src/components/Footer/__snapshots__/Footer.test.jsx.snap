// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="page-footer font-small pt-4 mt-4 footer"
  data-test="footer"
>
  <div
    className="container-fluid text-center text-md-left"
    data-test="container"
  >
    <div
      className="row"
      data-test="row"
    >
      <div
        className="col-md-6"
        data-test="col"
      >
        <h5
          className="title"
        >
          City Walks is...
        </h5>
        <p>
          (City Walks description)
        </p>
      </div>
      <div
        className="col-md-3"
        data-test="col"
      >
        <h5
          className="title"
        >
          Site Links
        </h5>
        <ul
          className="footer-ul"
        >
          <li
            className="list-unstyled"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="list-unstyled"
          >
            <a
              href="/cities"
              onClick={[Function]}
            >
              Cities
            </a>
          </li>
          <li
            className="list-unstyled"
          >
            <a
              href="/walks"
              onClick={[Function]}
            >
              Walks
            </a>
          </li>
          <li
            className="list-unstyled"
          >
            <a
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="list-unstyled"
          >
            <a
              href="/blog"
              onClick={[Function]}
            >
              Blog
            </a>
          </li>
          <li
            className="list-unstyled"
          >
            <a
              href="/contact"
              onClick={[Function]}
            >
              Contact
            </a>
          </li>
          <li
            className="list-unstyled"
          >
            <a
              href="/boards"
              onClick={[Function]}
            >
              User Login
            </a>
          </li>
        </ul>
      </div>
      <div
        className="col-md-3"
        data-test="col"
      >
        <h5
          className="title"
        >
          Extra Links
        </h5>
        <ul
          className="footer-ul"
        >
          <li
            className="list-unstyled"
          >
            <a
              href="/admin"
              onClick={[Function]}
            >
              Admin portal
            </a>
          </li>
          <li
            className="list-unstyled"
          >
            <a
              href="/privacy"
              onClick={[Function]}
            >
              Privacy Policy
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="footer-copyright text-center py-3"
  >
    <div
      className="container-fluid"
      data-test="container"
    >
      Â© 
      2021
       Copyright
    </div>
  </div>
</footer>
`;
